lib/Mojolicious/Guides/Cookbook.pod

```diff
@@ -65,6 +65,76 @@ connections concurrently.
 
 In L<Mojolicious> this event loop is L<Mojo::IOLoop>.
 
+=head2 Reverse Proxy
+
+A reverse proxy architecture is a deployment technique used in many production
+environments, where a I<reverse proxy> server is put in front of your
+application to act as the endpoint accessible by external clients. It can
+provide a lot of benefits, like terminating SSL connections from the outside,
+limiting the number of concurrent open sockets towards the Mojolicious
+application (or even using Unix sockets), balancing load across multiple
+instances, or supporting several applications through the same IP/port.
+
+   .               ..........................................
+                   :                                        :                        
+                   :                       +-------------+  :                        
+                   :  +------------+      +-------------+|  :
+   +--------+      :  |            |      |             ||  :
+   |        |-------->|  reverse   |----->| Mojolicious ||  :
+   | client |      :  |   proxy    |      | application ||  :
+   |        |<--------|            |<-----|             ||  :
+   +--------+      :  | (frontend) |      |  (backend)  ||  :
+                   :  |            |      |             |+  :
+                   :  +------------+      +-------------+   :                       
+                   :                                        :
+                   .. system boundary (e.g. same host) ......
+
+This setup introduces some problems, though: the application will receive
+requests from the reverse proxy instead of the original client; the
+address/hostname where your application lives internally will be different from
+the one visible from the outside; if terminating SSL, the reverse proxy exposes
```

Comment:
the one visible from the outside; and if terminating SSL, the reverse proxy exposes